src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -0,0 +1,1345 @@
+            String path = sb.ToString();
+            if (r == 0)
+                throw Win32Marshal.GetExceptionForLastWin32Error();
+            path = GetFullPathInternal(path);
+#if FEATURE_CORECLR
+#if !FEATURE_LEGACYNETCFIOSECURITY
+            FileSecurityState state = new FileSecurityState(FileSecurityStateAccess.Write, String.Empty, path);
+            state.EnsureState();
+#endif //!FEATURE_LEGACYNETCFIOSECURITY
+#endif
+            return path;
+        }
+
+        internal static bool IsRelative(string path)
+        {
+            Contract.Assert(path != null, "path can't be null");
+#if !PLATFORM_UNIX
+            if ((path.Length >= 3 && path[1] == VolumeSeparatorChar && path[2] == DirectorySeparatorChar &&
+                   ((path[0] >= 'a' && path[0] <= 'z') || (path[0] >= 'A' && path[0] <= 'Z'))) ||
+                  (path.Length >= 2 && path[0] == '\\' && path[1] == '\\'))
+#else
+            if(path.Length >= 1 && path[0] == VolumeSeparatorChar)
+#endif // !PLATFORM_UNIX
+                return false;
+            else
+                return true;
+        }
+
+        // Returns a cryptographically strong random 8.3 string that can be 
+        // used as either a folder name or a file name.
+        public static String GetRandomFileName()
+        {
+            // 5 bytes == 40 bits == 40/5 == 8 chars in our encoding
+            // This gives us exactly 8 chars. We want to avoid the 8.3 short name issue
+            IBuffer buffer = CryptographicBuffer.GenerateRandom(10);
+            byte[] key;
+            CryptographicBuffer.CopyToByteArray(buffer, out key);
+
+
+            // rndCharArray is expected to be 16 chars
+            char[] rndCharArray = Path.ToBase32StringSuitableForDirName(key).ToCharArray();
+            rndCharArray[8] = '.';
+            return new String(rndCharArray, 0, 12);
+        }
+
+        // Returns a unique temporary file name, and creates a 0-byte file by that
+        // name on disk.
+#if FEATURE_LEGACYNETCFIOSECURITY
+        [System.Security.SecurityCritical]
+#else
+        [System.Security.SecuritySafeCritical]
+#endif //FEATURE_LEGACYNETCFIOSECURITY
+        public static String GetTempFileName()
+        {
+            return InternalGetTempFileName(true);
+        }
+
+        [System.Security.SecurityCritical]
+        internal static String UnsafeGetTempFileName()
+        {
+            return InternalGetTempFileName(false);
+        }
+
+        [System.Security.SecurityCritical]
+        private static String InternalGetTempFileName(bool checkHost)
+        {
+            String path = GetTempPath();
+
+            // Since this can write to the temp directory and theoretically 
+            // cause a denial of service attack, demand FileIOPermission to 
+            // that directory.
+            StringBuilder sb = new StringBuilder(MAX_PATH);
+            uint r = Interop.mincore.GetTempFileNameW(path, "tmp", 0, sb);
```

Comment:
This is now going to regress our Unix implementation.  Previously Path was defined in mscorlib, and these "Win32" functions got routed through the PAL.  Now they're going to be P/Invokes to the Windows functions, which will blow up when we try to run this on Unix.